server/loadConfigs.js

```diff
@@ -27,6 +27,23 @@ function loadConfig(name, options = {}) {
   return config;
 }
 
+function getConfigFromOptions(options) {
+  let config;
+  if (!(options instanceof Config)) {
+    assert(options.id, 'must pass an id to test config options');
+    config = loadConfig(options.id, options);
+  } else {
+    assert(options.str('id'), 'must pass an id to test config options');
+    // making a copy from options and data properties
+    // to avoid any mutations
+    config = loadConfig(options.str('id'), {
+      ...options.options,
+      ...options.data
+    });
+  }
+  return config;
+}
+
```

Comment:
What do you think of this?

```js
function getConfigFromOptions(options) {
  if (!(options instanceof Config))
    options = loadConfig(options.id, options);
  assert(options.str('id'), 'must pass an id to test config options');
  // making a copy from options and data properties
  // to avoid any mutations
  return loadConfig(options.str('id'), {
      ...options.options,
      ...options.data
  });
}
```

Less `if/else` that you need to think about when reading it

Reply:
That could work. You lose the check for an `id` param before sending to `loadConfig` but it'll get caught in `loadConfig` eventually anyway, so not a big deal. Will update 👍

Reply:
Updated